### PR TITLE
Add default profile switching to konsole plugin

### PR DESCRIPTION
**Pull request by Seas0**
Making new tabs subject to changes in theme profile

### CODE REVIEW

**Comment by l0drex** [comment]
Works great in Yakuake, but it doesn't work in Konsole for me.

File: yin_yang/plugins/konsole.py [hunk 1]
```diff
@@ -266,3 +270,37 @@ def set_profile(service: str, profile: str):
         )
         message.setArguments([profile])
         connection.call(message)
+
+
+def set_default_profile(service: str, profile: str):
```
Comment (l0drex) [inline yin_yang/plugins/konsole.py:275]: This is really similar to `set_profile()`. Please merge the two into one function, f.e. `set_profile(default: bool)`

Comment (l0drex) [reply yin_yang/plugins/konsole.py:275]: Actually, we always call `set_profile()` and then `set_default_profile()`. So merge the two into one function, then call the function itself to change the default profile:

```python
def set_mode():
    set_profile('org.kde.konsole', profile)
    # ...

def set_profile(service: str, profile: str, default: bool = False):
    # ...
    if not default:
        set_profile(service, profile, True)
```

**Review by l0drex — CHANGES_REQUESTED** [review]
This looks promising! Some cleanup to do, and then we have to investigate what is missing for Konsole and Dolphin

File: yin_yang/plugins/konsole.py (hunk 1 above)
Comment (Seas0) [reply yin_yang/plugins/konsole.py]: Merging two functions totally make sense to me, so I have amended this PR for an extra argument in `set_profile`.
But the latter part, which suggested introducing a recursive calling, doesn't sound good to me, as it only reduces few lines of code but gains a lot more confusion for future maintenance.

**Review by l0drex — APPROVED** [review]
Thanks, looks good now!

**Pull request merged**